core/templates/pages/feedback-updates-page/feedback-updates-page.component.spec.ts

```diff
@@ -0,0 +1,848 @@
+        let NOW_MILLIS = 1617393321345;
+        spyOn(dateTimeFormatService, 'getLocaleAbbreviatedDatetimeString')
+          .withArgs(NOW_MILLIS).and.returnValue('4/2/2021');
+
+        expect(component.getLocaleAbbreviatedDatetimeString(NOW_MILLIS))
+          .toBe('4/2/2021');
+      });
+
+    it('should sanitize given png base64 data and generate url', () => {
+      let result = component.decodePngURIData('%D1%88%D0%B5%D0%BB%D0%BB%D1%8B');
+
+      fixture.detectChanges();
+
+      expect(result).toBe('шеллы');
+    });
+  });
+
+  describe('when fetching dashboard data fails', () => {
+    beforeEach(async(() => {
+      TestBed.configureTestingModule({
+        imports: [
+          BrowserAnimationsModule,
+          MaterialModule,
+          FormsModule,
+          HttpClientTestingModule
+        ],
+        declarations: [
+          FeedbackUpdatesPageComponent,
+          MockTranslatePipe,
+          SortByPipe,
+          MockSlicePipe,
+          MockTrunctePipe,
+          BackgroundBannerComponentStub,
+          LoadingDotsComponentStub,
+        ],
+        providers: [
+          AlertsService,
+          CsrfTokenService,
+          FeedbackUpdatesBackendApiService,
+          UserService,
+          PageTitleService,
+          {
+            provide: TranslateService,
+            useClass: MockTranslateService
+          }
+        ],
+        schemas: [NO_ERRORS_SCHEMA]
+      }).compileComponents();
+    }));
+
+    beforeEach(fakeAsync(() => {
+      fixture = TestBed.createComponent(FeedbackUpdatesPageComponent);
+      component = fixture.componentInstance;
+      alertsService = TestBed.inject(AlertsService);
+      csrfTokenService = TestBed.inject(CsrfTokenService);
+      feedbackUpdatesBackendApiService =
+        TestBed.inject(FeedbackUpdatesBackendApiService);
+      userService = TestBed.inject(UserService);
+      translateService = TestBed.inject(TranslateService);
+      pageTitleService = TestBed.inject(PageTitleService);
+
+      spyOn(csrfTokenService, 'getTokenAsync').and.returnValue(
+        Promise.resolve('sample-csrf-token'));
+
+      spyOn(userService, 'getProfileImageDataUrl').and.returnValue(
+        ['default-image-url-png', 'default-image-url-webp']);
+
+      spyOn(userService, 'getUserInfoAsync').and.returnValue(
+        Promise.resolve(userInfo as UserInfo));
+    }));
+
+    it('should show an alert warning when fails to get feedback updates data',
+      fakeAsync(() => {
+        const fetchDataSpy = spyOn(
+          feedbackUpdatesBackendApiService,
+          'fetchFeedbackUpdatesDataAsync')
+          .and.rejectWith(404);
+        const alertsSpy = spyOn(alertsService, 'addWarning').and.callThrough();
+
+        component.ngOnInit();
+
+        tick(1000);
+        fixture.detectChanges();
+
+        expect(alertsSpy).toHaveBeenCalledWith(
+          'Failed to get feedback updates data');
+        expect(fetchDataSpy).toHaveBeenCalled();
+        tick(1000);
```

Comment:
Add a comment explaining why a `tick` should be present here.

Reply:
I have added tick to remove the 2 timer error. But now I have used flush(); to remove it.